src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -209,12 +245,6 @@ int main(int argc, char *argv[])
     Log::Info << "Loaded kNN model from '" << inputModelFile << "' (trained on "
         << knn.Dataset().n_rows << "x" << knn.Dataset().n_cols << " dataset)."
         << endl;
-
-    // Adjust singleMode and naive if necessary.
-    knn.SingleMode() = CLI::HasParam("single_mode");
-    knn.Naive() = CLI::HasParam("naive");
-    knn.LeafSize() = size_t(lsInt);
-    knn.Epsilon() = epsilon;
```

Comment:
@rcurtin
I removed all that options because they are incorrect:
+ For example if we save a model in `SingleMode`, and we reload that model with no extra option, it will set:
```
  knn.SingleMode() = CLI::HasParam("single_mode") ; // false
  knn.Naive() = CLI::HasParam("naive")  ;  // false
```
  It will do `Dual Tree Search` instead of `Single Tree Search`.
+ Also, if we set the `LeafSize()` to a different value, we need to call `BuildModel` again, to rebuild the reference tree.
+ Also, in the parameter specifications, it is mentioned that "--leaf_size" "--naive" and "--single_mode" would be ignored when providing an input model file. So, I wanted to be consistent with that comments.

So, it is a bit difficult to properly update a given input model.
For example, if users provide an input model file, without the options "--naive" and "--single_mode", does it means they don't want to modify the model or that they want to update it to do "dual tree search"?